File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(a,b):
    answer = 0
    for i in range(a, b+1, 1):
        c = [x for x in range(1,i+1) if i % x == 0]
        if len(c) % 2 == 0:
            answer += i
        else:
            answer -= i
    return answer

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.swift
import Foundation

func solution(_ left:Int, _ right:Int) -> Int {
    var answer : Int = 0
    var count : Int = 0
    for num in left...right {
        count = 0
        for index in 1...num {
            if num % index == 0 {
                count += 1
            }
        }
        if (count % 2 == 0) && (count != 0) {
            answer += num
        }else if (count % 2 != 0) {
            answer -= num
        }
    }
    return answer
}